### ISSUE TITLE
0.1.11 caused fatal error: socket hang up

### ISSUE THREAD

**Issue by asujata** [drewzboto/grunt-connect-proxy#70]
The latest update to 0.1.11 caused my existing proxy setup to fail with "fatal error: socket hang up".

My setup:

```
proxies: [
          {
            context: '/api',
            host: '...',
            port: 443,
            https: true,
            changeOrigin: true,
            xforward: false,
            rejectUnauthorized: false
          }
        ]
```

The error only occured when trying to use https.

Switching back to 0.1.10 resolved my issue.

**Comment by procyborg** [comment]
absolutely. reverting to 0.1.10 suppresses the socket-hangup bug. i am proxying a https endpoint for oauth2.
thanks for the pointer guys.

**Comment by ihr** [comment]
I can confirm that changing the port worked for me. Here is relevant part of my **Gruntfile.js**:

```
proxies: [
                {
                    context: ['/api'],
                    host: 'localhost',
                    port: 8090,
                    https: false,
                    rewrite: {
                        '^/api': ''
                    }
                }
            ]
```

**Comment by bostrom** [comment]
I'm having a problem with secure WebSocket connections over the proxy (HTTPS->HTTP). Normal websocket connections work fine in my app. So I thought I'd debug the wss case and wrote a WebSocket test for it. I decided to go simple and started off with just normal ws:// connections. 

However, even that didn't work since I got slapped in the face by this "fatal error: socket hang up" bug, so now I'm stuck with no working websocket connections over the proxy whatsoever (even though they work in my browser). Changing ports doesn't seem to work and in 0.1.10 the tests won't even run. I've tried to dig in to http-proxy also, but to no avail.

The branch with the websocket test is here https://github.com/Flexim/grunt-connect-proxy/tree/wss if you're interested in debugging it with me.

**Comment by xiaoshao** [comment]
I have the same problem in 0.1.11. I have two proxies as follows.
`{
                context: '/google',
                host: 'google.com',
                port: '80',
                rewrite: {
                    '^/google': '',
                }
            },
            {
                context: '/baidu',
                host: 'baidu.com',
                port: '80',
                rewrite: {
                    '^/baidu': '/',
                }
            }`
the proxy for google works well. but the proxy for baidu doesn't work.

**Comment by ianblenke** [comment]
I'm having the same "Fatal error: socket hang up" problem with https proxy. Using http works just fine.  Reverting to 0.1.10 doesn't seem to help. Would love to turn up the debugging here to understand what the problem may be. Any hints as to how to accomplish that?

**Comment by ianblenke** [comment]
Correction: reverting to 0.1.10 did work, I had to clean out the newer node_modules version first.

**Comment by dzdrazil** [comment]
I'm also having this problem on 0.2.0 (win8, node 0.12.2); reverting to 0.1.10 did work for me as well.

**Comment by lostthetrail** [comment]
I confirm this is still happening on 0.2.0

**Comment by mteodori** [comment]
same issue here, and the servers I am proxying to have valid certificates, not self-signed ones, the only solution is to stick to grunt-connect-proxy@0.1.10 and block it via npm shrinkwrap --dev
